### PR TITLE
MonaiAlgo: fix logging in multi-gpu training

### LINKED ISSUE #5354: Reduce redundant logging in multi-gpu training with MonaiAlgo
**Is your feature request related to a problem? Please describe.**
Currently, each rank is logging during multi-gpu training.

**Describe the solution you'd like**
Remove redundant logs.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by holgerroth**
Signed-off-by: Holger Roth <hroth@nvidia.com>

Fixes #5354.

### Description

Previous output:
```
Current output:
2022-10-18 12:45:59,679 - MonaiAlgo - INFO - Using multi-gpu training on rank 1 (available devices: 2)
2022-10-18 12:45:59,681 - MonaiAlgo - INFO - Using multi-gpu training on rank 0 (available devices: 2)
2022-10-18 12:49:48,790 - ignite.engine.engine.SupervisedTrainer - INFO - Got new best metric of train_accuracy: 0.802879669048168
2022-10-18 12:49:48,790 - ignite.engine.engine.SupervisedTrainer - INFO - Got new best metric of train_accuracy: 0.802879669048168
2022-10-18 12:49:56,579 - ignite.engine.engine.SupervisedEvaluator - INFO - Got new best metric of val_mean_dice: 0.1470419466495514
2022-10-18 12:49:56,579 - ignite.engine.engine.SupervisedEvaluator - INFO - Got new best metric of val_mean_dice: 0.1470419466495514
```
Output after fix:
```
2022-10-18 12:51:05,400 - MonaiAlgo - INFO - Using multi-gpu training on rank 0 (available devices: 2)
2022-10-18 12:51:05,410 - MonaiAlgo - INFO - Using multi-gpu training on rank 1 (available devices: 2)
2022-10-18 12:53:09,889 - ignite.engine.engine.SupervisedTrainer - INFO - Got new best metric of train_accuracy: 0.6750877521656178
2022-10-18 12:53:25,170 - ignite.engine.engine.SupervisedEvaluator - INFO - Got new best metric of val_mean_dice: 0.06980131566524506
```

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/fl/client/monai_algo.py [hunk 1]
```diff
@@ -494,6 +494,11 @@ def initialize(self, extra=None):
             BundleKeys.DATASET_DIR, default=ConfigItem(None, BundleKeys.DATASET_DIR)
         )
 
+        if self.multi_gpu:
```
Comment (wyli) [inline monai/fl/client/monai_algo.py:497]: is it possible to achieve this in https://github.com/Project-MONAI/model-zoo/blob/dev/models/spleen_ct_segmentation/configs/logging.conf cc @Nic-Ma

Comment (wyli) [reply monai/fl/client/monai_algo.py:497]: Ok, lets merge this for now but ideally the detailed logging logic should be agnostic to the trainer..

**Review by wyli — APPROVED** [review]

File: monai/fl/client/monai_algo.py (hunk 1 above)
Comment (ericspod) [reply monai/fl/client/monai_algo.py:497]: There's ways of changing the logger's [Formatter class](https://docs.python.org/3/library/logging.html#logging.Formatter) to write the process and thread ID with each message. This doesn't state explicitly which rank you're on but you can work it out from the PID values or at least know which process is which.

Comment (Nic-Ma) [reply monai/fl/client/monai_algo.py:497]: I feel it's hard for common users to find out the matching process ID of ranks..
CC @holgerroth @wyli .

Thanks.

Comment (ericspod) [reply monai/fl/client/monai_algo.py:497]: It is so you're not really finding that out, just differentiating which process wrote which line in the log, though typically the children PIDs  will be only a few values more than the parent's PID and in order.

**Pull request merged**